tests/testthat/test-subset_draws.R

```diff
@@ -134,3 +134,14 @@ test_that("subset_draws preserves variable order", {
   expect_equal(variables(x), c("theta[2]", "theta[1]"))
 })
 
+test_that("non-unique subsetting for draws_df same as doing it with draws_list", {
+  x_df <- as_draws_df(example_draws())
+  x_list <- as_draws_list(example_draws())
+
+  x_df_sub <- subset_draws(x_df, chain = c(1,1,2), iteration = c(1:2, 1:50),
+                           unique = FALSE)
+  x_list_sub <- subset_draws(x_list, chain = c(1,1,2), iteration = c(1:2, 1:50),
+                             unique = FALSE)
+  expect_equivalent(x_df_sub, as_draws_df(x_list_sub))
```

Comment:
@paul-buerkner I had to use expect_equivalent here instead of expect_equal because the `as_draws_df(x_list_sub)` doesn't have row names. Do you know if that's intentional? It's only an issue after subsetting. Check this out: 

```r
x_df <- as_draws_df(example_draws())
x_list <- as_draws_list(example_draws())

# all good here
all.equal(rownames(x_df), rownames(as_draws_df(x_list))) 

x_df_sub <- subset_draws(x_df)          
x_list_sub <- subset_draws(x_list)

# now no row names for the second one
all.equal(rownames(x_df_sub), rownames(as_draws_df(x_list_sub)))
```

Reply:
I don't think its intentional. in posterior, we don't really care about the rownames anywhere so I don't think it matter practically, but of course esthetically, a consistent behavior is desirble.

Reply:
I am fine with sorting out the row names thing separately.

Reply:
Ok let's go ahead and merge this and I'll open a separate issue about the row names